envelopPlugins.ts

```diff
@@ -0,0 +1,78 @@
+import { DefaultContext } from "@envelop/core";
+import { useGenericAuth } from "@envelop/generic-auth";
+import * as jose from "jose";
+
+type UserType =
+  | {
+      id: string;
+      roles: string[];
+    }
+  | {};
+
+const VALID_ROLES = ["admin", "employee", "buyer"];
+
+/**
+ * Used for JWT verification
+ */
+const JWKS = jose.createRemoteJWKSet(
+  new URL(
+    "http://localhost:3500/v1.0/invoke/keycloak/method/keycloak/realms/Misarch/protocol/openid-connect/certs"
+  )
+);
+
+/**
+ * Resolves the user from the JWT token
+ *
+ * @param context GraphQL envelop context used to access request headers
+ * @returns The user object if the token is valid, null otherwise
+ */
+async function resolveUserAuthenticated(
+  context: DefaultContext
+): Promise<UserType | null> {
+  try {
+    const authHeader = (context.request as any)?.headers?.get("authorization");
+    if (authHeader == undefined) {
+      return {};
+    }
+    let parsedHeader: string;
+    if (authHeader.startsWith("Bearer ")) {
+      parsedHeader = authHeader.substring(7);
+    } else {
+      parsedHeader = authHeader;
+    }
+    const { payload } = await jose.jwtVerify(parsedHeader, JWKS, {});
+    const user: UserType = {
+      id: payload.sub!,
+      roles: getValidRoles(payload),
+    };
+    context.authHeader = parsedHeader;
```

Comment:
Wait, without the "Bearer"?
Why?
This sounds like a bad idea for the sake of removing a few lines.

Reply:
changed with 7c70647811b735464e80c89c6947bf49fff5d0c5